File: 백준/Silver/1764. 듣보잡/듣보잡.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");
let [a , b] = input[0].split(" ").map(Number);

let map1 = new Set();
let map2 = new Set();

for(let i = 1; i <= input.length; i++){
  if(i <= a) map1.add(input[i]);
  else map2.add(input[i]);
}

let result = [...map1].filter((data) => map2.has(data)).sort();
console.log(result.length + "\n" + result.join("\n"));

File: 백준/Silver/1764. 듣보잡/듣보잡.py
import sys
input = sys.stdin.readline

N , M = map(int , input().strip().split())

arr1 = [input().rstrip() for _ in range(N)]
arr2 = [input().rstrip() for _ in range(M)]


result = []

arr_dict = dict()

for i in arr1:
    arr_dict[i] = 1
    
for i in arr2:
    if i in arr_dict:
        result.append(i)
        
result.sort()
print(len(result))
for i in result:
    print(i)
